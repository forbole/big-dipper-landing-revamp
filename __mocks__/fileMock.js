// __mocks__/fileMock.js
module.exports = {
  blurDataURL: 'data:image/png;base64,imagedata',
  height: 24,
  src: '/img.jpg',
  width: 24,
}