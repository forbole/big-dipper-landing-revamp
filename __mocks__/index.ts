export { default as MockTheme } from "./theme";
