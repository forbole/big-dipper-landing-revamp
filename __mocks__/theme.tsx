/* eslint-disable */
import { ThemeProvider } from '@mui/material/styles';
import { ReactNode } from 'react';
import { darkTheme } from '@/src/styles/theme';

/**
 * Theme mocker to handle custom keys
 */
const MockTheme = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={darkTheme}>{children}</ThemeProvider>;
};

export default MockTheme;
