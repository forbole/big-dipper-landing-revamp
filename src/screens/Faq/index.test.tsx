/* eslint-disable react/display-name */
import Box from '@mui/material/Box';
import { act, render, screen } from '@testing-library/react';
import { MockTheme } from '@/__mocks__';
import Component from '.';

const mockI18n = {
  t: (key: string) => key,
  lang: 'en',
};
jest.mock('next-translate/useTranslation', () => () => mockI18n);
jest.mock('@/src/components/Layout', () => (props: object) => (
  <Box data-testid="Layout" {...props} />
));
jest.mock('@/src/components/SectionBox', () =>
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  ({ main, ...props }: { [p: string]: unknown }) => (
    <Box data-testid="SectionBox" {...props} />
  )
);
jest.mock('@/src/components/SectionLimit', () => (props: object) => (
  <Box data-testid="SectionLimit" {...props} />
));
jest.mock('@/src/components/ContentBox', () => (props: object) => (
  <Box data-testid="ContentBox" {...props} />
));

jest.mock(
  './components/MenuDesktop',
  () =>
    ({ handleChange: _, ...props }: { [p: string]: unknown }) =>
      <div data-testid="MenuDesktop" {...props} />
);
jest.mock(
  './components/MenuMobile',
  () =>
    ({ handleChange: _, ...props }: { [p: string]: unknown }) =>
      <Box data-testid="MenuMobile" {...props} />
);
jest.mock(
  './components/Content',
  () =>
    ({ handleChange: _, ...props }: { [p: string]: unknown }) =>
      <Box data-testid="Content" {...props} />
);
// ==================================
// unit tests
// ==================================
describe('FAQ', () => {
  it('matches snapshot', async () => {
    await act(async () => {
      global.innerWidth = 1400;
      global.dispatchEvent(new Event('resize'));
    });

    const { container } = render(
      <MockTheme>
        <Component />
      </MockTheme>
    );

    expect(screen.getByTestId('Layout')).toBeInTheDocument();
    expect(screen.getByTestId('SectionBox')).toBeInTheDocument();
    expect(screen.getByTestId('SectionLimit')).toBeInTheDocument();
    expect(screen.getByTestId('MenuDesktop')).toBeInTheDocument();
    expect(screen.getByTestId('Content')).toBeInTheDocument();
    expect(screen.getByTestId('MenuDesktop').hasAttribute('items')).toBe(true);

    expect(container).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
