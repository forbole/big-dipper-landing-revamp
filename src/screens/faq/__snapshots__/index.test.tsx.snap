// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQ matches snapshot 1`] = `
.emotion-0 .title {
  color: #fff;
}

.emotion-0 .header__wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  margin-bottom: 16px;
}

.emotion-0 .content__wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-1 {
  margin: 0;
  font-size: 2rem;
  font-weight: 600;
  font-family: SFPro,Arial,sans-serif;
  line-height: 1.2;
}

@media (min-width:1280px) {
  .emotion-2 {
    display: none;
  }
}

.emotion-3 {
  display: none;
}

@media (min-width:1280px) {
  .emotion-3 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    margin-right: 32px;
  }
}

.emotion-4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div>
  <div
    data-testid="Layout"
  >
    <div
      data-testid="SectionBox"
    >
      <div
        class="emotion-0"
        data-testid="SectionLimit"
      >
        <div
          class="header__wrapper"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 title emotion-1"
          >
            faq
          </h2>
          <div
            class="emotion-2"
            data-testid="MenuMobile"
            items="blockchain.title,stakers.title,validators.title,governance.title"
          />
        </div>
        <div
          class="content__wrapper"
        >
          <div
            class="emotion-3"
            data-testid="MenuDesktop"
            items="blockchain.title,stakers.title,validators.title,governance.title"
          />
          <div
            class="emotion-4"
            content="[object Object],[object Object]"
            data-testid="Content"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
